src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [state];
+
+  for (let i = 0; i < actions.length; i++) {
+    const prevState = result[i];
```

Comment:
why is it here?
you need to keep your `currentState` outside loop

`let currentState = { ...state }`